lldb/source/Plugins/ObjectFile/Mach-O/ObjectFileMachO.cpp

```diff
@@ -6519,15 +6519,13 @@ struct page_object {
 };
 
 bool ObjectFileMachO::SaveCore(const lldb::ProcessSP &process_sp,
-                               const FileSpec &outfile,
-                               lldb::SaveCoreStyle &core_style, Status &error) {
-  if (!process_sp)
+                               const lldb_private::CoreDumpOptions &core_options,
+                               Status &error) {
+  auto core_style = core_options.GetCoreDumpStyle();
+  const auto outfile = core_options.GetOutputFile();
```

Comment:
We already check if the FileSpec is valid up in in `PluginManager::SaveCore(...)` so we can just get a valid FileSpec here and unrwap the optional. Add an assert() since this is checked:
```
// The FileSpec is already checked in PluginManager::SaveCore.
assert(core_options.GetOutputFile().has_value()); 
const FileSpec &outfile = core_options.GetOutputFile().value();
```